plugins/tokens/index.js

```diff
@@ -0,0 +1,96 @@
+//@ts-check
+const sassExtract = require('sass-extract');
+const glob = require('fast-glob');
+const fs = require('fs-extra');
+const path = require('path');
+const { getFinalFilePart, convertSassVariable } = require('./conversation');
+
+const distFolder = path.join(process.cwd(), '/dist');
+
+createTokens(distFolder);
+
+async function createTokens(distFolder) {
+	// Global variables play better when we just import a single index file
+	const entries = await glob(['../../css/src/tokens/index.scss'], {});
+
+	if (entries.length !== 1) {
+		throw new Error(
+			`Error with entries. There should only be a single entry. Got ${entries.length}.`
+		);
+	}
+
+	await fs.mkdir(distFolder);
+
+	sassExtract
+		.render({
+			file: entries[0]
+		})
+		.then(async rendered => {
+			const tokens = collectTokenSources(rendered);
+			writeVariables(rendered, tokens);
+
+			Object.keys(tokens).map(token => {
+				const outfileStem = path.join(distFolder, token);
+				const json = JSON.stringify(tokens[token]);
+				// writing json output
+				fs.writeFile(outfileStem + '.json', json);
```

Comment:
super nit- fs-extra has writeJson `await fs.writeJson('./dist/index.json', tokens)`

Reply:
done, though with quicktype needing a string it doesn't matter much